### ISSUE TITLE
tensorflow.keras: evaluate with custom loss and metric gives wrong output

### ISSUE THREAD

**Issue by corner100** [tensorflow/tensorflow#24232]
This issue is about tensorflow.keras

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): pip virtual env
- TensorFlow version (use command below):('v1.11.0-0-gc19e29306c', '1.11.0')
- Python version: 2.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory: Geforce 940mx

**Describe the current behavior**
I have a custom metric
`def root_mean_squared_error(y_true, y_pred):
    return K.sqrt(K.mean(K.square(y_pred - y_true)))`
while training I get the following output:

> Train on 161532 samples, validate on 40385 samples
> Epoch 1/5
>  - 132s - loss: 3.6781 - root_mean_squared_error: 3.6781 - val_loss: 3.6463 - val_root_mean_squared_error: 3.6463
> Epoch 2/5
>  - 129s - loss: 3.6528 - root_mean_squared_error: 3.6528 - val_loss: 3.6287 - val_root_mean_squared_error: 3.6287
> Epoch 3/5
>  - 143s - loss: 3.6352 - root_mean_squared_error: 3.6352 - val_loss: 3.6210 - val_root_mean_squared_error: 3.6210
> Epoch 4/5
>  - 134s - loss: 3.6223 - root_mean_squared_error: 3.6223 - val_loss: 3.6369 - val_root_mean_squared_error: 3.6369
> Epoch 5/5

after running `model.evaluate(val_data, val_label, verbose = 2)`
I get the following output:
> [3.028181584421266, 3.028181584421266]
for `model.metrics_names` 
> ['loss', 'root_mean_squared_error']

after running
`from sklearn.metrics import mean_squared_error
np.sqrt(mean_squared_error(val_label,np.squeeze(model.predict(val_data, batch_size=256))))`
I get
> 3.7112323807806966`

**Describe the expected behavior**
all outputs should be the same

My model looks like this:

```
def get_model(input_size):
    input1 = tf.keras.layers.Input(shape=(input_size,))
    x1 = Dense(input_size, activation='elu')(input1)  # input_size heisst, so viele neurons wie inputs

    x2 = Dense(300, activation='relu')(x1)
    concat1 = tf.keras.layers.Concatenate(axis=-1)([x1, x2])
    x3 = Dense(300, activation='relu')(concat1)
    concat2 = tf.keras.layers.Concatenate(axis=-1)([concat1, x3])
    x4 = Dense(300, activation='relu')(concat2)
    concat3 = tf.keras.layers.Concatenate(axis=-1)([concat2, x4])
    x5 = Dense(300, activation='relu')(concat3)
    concat4 = tf.keras.layers.Concatenate(axis=-1)([concat3, x5])
    x6 = Dense(300, activation='relu')(concat4)

    # output Layer
    output = Dense(1, activation='linear')(x6)
    model = tf.keras.models.Model(inputs=input1, outputs=output)
    my_opt = tf.keras.optimizers.Adam(lr=0.001, beta_1=0.9, beta_2=0.999, epsilon=None, decay=0.0, amsgrad=False)
    my_loss = root_mean_squared_error
    model.compile(optimizer=my_opt, loss=my_loss,
                  metrics=[root_mean_squared_error])
    return model
```
===========================

If I run the exact same thing with
`def my_mean_squared_error(y_true, y_pred):
    return K.mean(K.square(y_pred - y_true))`
and
`model.compile(optimizer=my_opt, loss=my_loss,
                  metrics=[my_mean_squared_error,"mse"])`
and
`mean_squared_error(val_label,np.squeeze(model.predict(val_data, batch_size=256)))`
it works fine.

so it seems, that there is something wrong with K.sqrt()

**Comment by ymodak** [comment]
Apologies for the delay in response. I created a short code snippet to compare the sqrt results by using tf.keras, keras and numpy. The results are identical and correct.
``` python
import tensorflow as tf
from keras import backend as K
import numpy as np
x = tf.constant(9878389478434.12)
sess = tf.Session()
x_1 = sess.run(x)
print('tf.sqrt(x)= ' + str(sess.run(tf.sqrt(x))))
print('K.sqrt(x)= ' + str(sess.run(K.sqrt(x))))
print('np.sqrt(x_1)= ' + str(np.sqrt(x_1)))
```
Output:
```python
tf.sqrt(x)= 3142990.5
K.sqrt(x)= 3142990.5
np.sqrt(x_1)= 3142990.5
```

**Comment by ymodak** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
